include/inviwo/core/datastructures/isovaluecollection.h

```diff
@@ -0,0 +1,170 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#ifndef IVW_ISOVALUECOLLECTION_H
+#define IVW_ISOVALUECOLLECTION_H
+
+#include <inviwo/core/common/inviwocoredefine.h>
+#include <inviwo/core/common/inviwo.h>
+
+#include <inviwo/core/datastructures/isovalue.h>
+#include <inviwo/core/util/observer.h>
+#include <inviwo/core/util/fileextension.h>
+
+#include <ostream>
+
+namespace inviwo {
+
+class IVW_CORE_API IsoValueCollectionObserver : public Observer {
+public:
+    virtual void onIsoValueAdded(IsoValue* v);
+    virtual void onIsoValueRemoved(IsoValue* v);
+    virtual void onIsoValueChanged(const IsoValue* v);
+};
+class IVW_CORE_API IsoValueCollectionObservable : public Observable<IsoValueCollectionObserver> {
+protected:
+    void notifyIsoValueAdded(IsoValue* v);
+    void notifyIsoValueRemoved(IsoValue* v);
+    void notifyIsoValueChanged(const IsoValue* v);
+};
+
+/**
+ * \ingroup datastructures
+ * \class IsoValueCollection
+ * \brief data structure managing multiple isovalues
+ *
+ * \see IsoValue
+ */
+class IVW_CORE_API IsoValueCollection : public Serializable,
+                                        public IsoValueCollectionObservable,
+                                        public IsoValueObserver {
+public:
+    IsoValueCollection(const std::vector<IsoValueData>& values = {});
+    IsoValueCollection(const IsoValueCollection& rhs);
+    virtual ~IsoValueCollection() = default;
+
+    IsoValueCollection& operator=(const IsoValueCollection& rhs);
+
+    size_t getNumIsoValues() const;
+
+    IsoValue* getIsoValue(size_t i);
+    const IsoValue* getIsoValue(size_t i) const;
+
+    std::vector<IsoValueData> getIsoValues() const;
```

Comment:
There are use cases for both. E.g. the TF editor requires access to the unsorted isovalues since the order is stable. Sorted ones could be used for output etc.